### ISSUE TITLE
[coqpp] [dune] coqpp locations don't show directory of file properly

### ISSUE THREAD

**Issue by ejgallego** [coq/coq#8772]
When compiling with Dune, see for example this backtrace:
```
Called from file "g_ltac.mlg", line 391, characters 6-82
Called from file "proofs/proof_global.ml", line 170, characters 24-36
```
the file should be `plugins/ltac/g_ltac.mlg`

**Comment by ejgallego** [comment]
This is solved in Dune 1.10, we can close this issue when we upgrade and update our `dune` files to use the `coq.pp` stanza.

**Comment by ejgallego** [comment]
Should be fixed by #10515 , however in composed builds it seems some problems do still happen.

**Comment by ejgallego** [comment]
Seems fixed in Dune 2.9 ; not explicitly but likely due to a better API for path manipulation and the Coq mode adopting it.

[The example was for example building serapi with Coq vendored]
